constellation-app/constellation
# Feature/issue#34

### Description of the Change

Modifed handling of date time import dialog to ensure CUSTOM format text field is enabled and editable ONLY when CUSTOM format is selected.
This change ensures both editable and enabled values of the widget(s) are kept in synch so that the text field and dropdown are only available/editable when required. Previously they were falling out of synch.

### Alternate Designs

No - this is addressing a bug with limited scope.

### Why Should This Be In Core?

Address identified bug.

### Benefits

Ability to set custom format as required. Removal of some raw exceptions being thron for incorrect datetime format strings.

### Possible Drawbacks

The original code, had a comment in ValueInputPane (common code) concerned with abiltiy to read greyed out text (I have no problem reading this), as well as indicating a disabled widget disables scrollbar, in discussiuons we decided to allow this to be the case.

### Verification Process

1. Open new graph.
2. Select Import->From Delimited File ....
3. Select file to import containing a timestamp and drag DateTime from Transaction Attributes column onto it.
4. Right click on Datetime and select Formatter->Datetime.
_Confirm that the Custom Format text entry field is disabled and unable to be selected for edit_
5. Modify the Datetime Format value to CUSTOM.
_Confirm that the Custom Format text entry field is enabled and able to be selected for edit_
6. Change the Datetime Format value a few times including some changes to and from CUSTOM as a value.
_Confirm that the Custom Format text entry field is enabled and able to be selected for edit ONLY when CUSTOM is set as a format_
7. Modify the Datetime Format value to CUSTOM.
8. Enter a valid custom format in line with the raw data and press the OK button.
_Confirm the values are shown with correct formatting_
9. Right click on Datetime and select Formatter->Datetime.
_Confirm the previously set CUSTOM format is displayed and is editable._
10. Repeat step 6 and confirm behaviour is still correct.
**Now testing exception handlling as discussed in #217 **
11. Select CUSTOM format and enter yyyyyyyyyyyyyyyyyyyy.
_Confirm no exception is thrown but datetime values all show with text = ERROR_
12. Right click on Datetime and select Formatter->Datetime.
_Confirm the previously set CUSTOM format is displayed and is editable._
13. Select CUSTOM format and enter aa.
_Confirm no exception is thrown but datetime values all show with text = ERROR_

### Applicable Issues

#34 
#217
